### PR TITLE
Component namespace and theme augmentation 

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/next/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/material-ui-lab/src/themeAugmentation/props.d.ts [hunk 1]
```diff
@@ -34,6 +42,25 @@ export interface LabComponentsPropsList {
   MuiToggleButtonGroup: ToggleButtonGroupProps;
   MuiTreeItem: TreeItemProps;
   MuiTreeView: TreeViewProps;
+  MuiPickersDay: PickersDayProps;
+  MuiDayPicker: DayPickerProps;
+  MuiMonthPicker: MonthPickerProps;
+  MuiPickersCalendarSkeleton: CalendarSkeletonProps;
```
Comment (oliviertassinari) [inline packages/material-ui-lab/src/themeAugmentation/props.d.ts:48]: ?
```suggestion
  MuiCalendarSkeleton: CalendarSkeletonProps;
```

File: packages/material-ui-lab/src/themeAugmentation/overrides.d.ts [hunk 2]
```diff
@@ -34,6 +47,35 @@ export interface LabComponentNameToClassKey {
   MuiToggleButtonGroup: ToggleButtonGroupClassKey;
   MuiTreeItem: TreeItemClassKey;
   MuiTreeView: TreeViewClassKey;
+  // pickers sources
+  MuiCalendar: Classes<typeof import('../DayPicker/Calendar').styles>;
```
Comment (oliviertassinari) [inline packages/material-ui-lab/src/themeAugmentation/overrides.d.ts:51]: I think that we will need to reconsider this name, especially if we go into http://fullcalendar.io/ land.

Comment (dmtrKovalenko) [reply packages/material-ui-lab/src/themeAugmentation/overrides.d.ts:51]: `MuiPickersClock` | `MuiPickersCalendarHeader` sounds better? What to do with `MuiClock` and `MuiClockNumber`?

Comment (oliviertassinari) [reply packages/material-ui-lab/src/themeAugmentation/overrides.d.ts:51]: For the Calendar, I think that we should consider `PickersCalendar`.

Can you think of a component that might conflict with Clock in the future? Personally, I don't, maybe it's fine?

File: packages/material-ui-lab/src/themeAugmentation/props.d.ts (hunk 1 above)
Comment (oliviertassinari) [reply packages/material-ui-lab/src/themeAugmentation/props.d.ts:48]: The other way around sounds fine too

```suggestion
MuiPickersCalendarSkeleton: PickersCalendarSkeletonProps;
```

**Pull request merged**